### PR TITLE
Improve crm_mon output when guest node's container and remote resource are on different nodes

### PR DESCRIPTION
**Pull request by nrwahl2**
In this situation, currently the GuestOnline line shows `<guest node>@<connection's node>`. It should show `<guest node>` when `--show-detail` is not set, and `<guest node>@<container's node>` when `--show-detail` is set.

Closes T180
Closes CLBZ#5373

### CODE REVIEW

File: cts/cts-cli.in [hunk 1]
```diff
@@ -544,6 +544,20 @@ function test_crm_mon() {
 
     rm -r "$CIB_file"
     unset CIB_file
+
+    export CIB_file="$test_home/cli/crm_mon-T180.xml"
+
+    desc="Text output of guest node's container on different node from its"
+    dest="$desc remote resource"
```
Comment (kgaillot) [inline cts/cts-cli.in:551]: presumably s/dest/desc/ here and below

**Review by kgaillot — COMMENTED** [review]
looks good

File: cts/cts-cli.in (hunk 1 above)
Comment (nrwahl2) [reply cts/cts-cli.in:551]: Yep, thanks

**Comment by nrwahl2** [comment]
The CI failures are things that don't match the commits... it's comparing against the old `regression.crm_mon.exp` instead of the new one. A retest might solve that.

However, I see that the `rpm` target fails on my machine. The new `cts-cli` test uses `failed_action_friendly()` when doing a regular source run, and it uses `failed_action_technical()` when run as part of an RPM build.
```
--- /tmp/cts-cli.crm_mon.Ypskjhwdbb     2022-09-15 11:59:31.188952362 -0700
+++ cts/cli/regression.crm_mon.exp      2022-09-15 11:49:15.000000000 -0700
@@ -4269,7 +4269,7 @@
     * httpd-bundle2-0 (192.168.20.190) (ocf:heartbeat:apache):  Started cent7-host2
 
 Failed Resource Actions:
-  * httpd-bundle1-0_monitor_60000 on cent7-host1 'error' (1): call=9, status='Error', queued=0ms, exec=0ms
+  * httpd-bundle1-0 60s-interval monitor on cent7-host1 could not be executed (Error) at Wed Dec  5 20:23:49 2018
```

It's because the RPM build defines `PCMK__COMPAT_2_0`, and when that's defined, `crm_mon` sets `pcmk_show_failed_detail` automatically. I'm checking whether removing the failed actions altogether solves it for this case. I'm surprised this doesn't break any existing tests.

**Comment by kgaillot** [comment]
> It's because the RPM build defines `PCMK__COMPAT_2_0`, and when that's defined, `crm_mon` sets `pcmk_show_failed_detail` automatically. I'm checking whether removing the failed actions altogether solves it for this case. I'm surprised this doesn't break any existing tests.

Our typical approach is to always pass --show-detail so the output is the same regardless of configure options

**Pull request merged**